src/pudl/transform/eia860.py

```diff
@@ -1157,3 +1157,172 @@ def cooling_equipment_continuity(cooling_equipment):
+    # Every other year 2009-2012 "01" refers to 2009 operating FGD equipment, and "1"
+    # refers to the 1976 operating FGD equipment. In 2011 the plant reports two "1"
+    # plants, so we change the SO2 control ID to "01" for the 2009 unit to preserve
+    # uniqueness.
+    fgd_df.loc[
+        (fgd_df.plant_id_eia == 6016)
+        & (fgd_df.report_date == "2011-01-01")
+        & (fgd_df.fgd_operating_date == "2009-03-01"),
+        "so2_control_id_eia",
+    ] = "01"
+
+    # Add a manufacturer name from the code.
+    fgd_df["fgd_manufacturer"] = fgd_df.fgd_manufacturer_code.map(
+        pudl.helpers.label_map(
+            CODE_METADATA["core_eia__codes_environmental_equipment_manufacturers"][
+                "df"
+            ],
+            from_col="code",
+            to_col="description",
+            null_value=pd.NA,
+        )
+    )
+
+    # Check for uniqueness of index and handle special case duplicates.
+    pkey = ["plant_id_eia", "so2_control_id_eia", "report_date"]
+    fgd_df = pudl.helpers.dedupe_and_drop_nas(fgd_df, primary_key_cols=pkey)
+
+    # After 2012, treated wastewater and water get lumped into what was previously
+    # reported as just a water code. Let's rename this code to "W" to preserve detail in
+    # earlier years.
+    for sorbent_col in (col for col in fgd_df.columns if "sorbent_type" in col):
+        fgd_df.loc[fgd_df[sorbent_col] == "WT", sorbent_col] = "W"
+
+    # Handle some non-numeric values in the pond landfill requirements column
+    fgd_df["pond_landfill_requirements_acre_foot_per_year"] = pd.to_numeric(
+        fgd_df.pond_landfill_requirements_acre_foot_per_year, errors="coerce"
+    )
+
+    fgd_df = (
+        pudl.metadata.classes.Package.from_resource_ids()
+        .get_resource("_core_eia860__fgd_equipment")
+        .encode(fgd_df)
+    )
+
+    return fgd_df.pipe(apply_pudl_dtypes, strict=True)
+
+
+@asset_check(asset=_core_eia860__fgd_equipment, blocking=True)
+def fgd_equipment_null_check(fgd):
+    """Check that columns other than expected columns aren't null."""
+    fast_run_null_cols = {
+        "county",
+        "fgd_operational_status_code",
+        "operating_date",
+        "fgd_manufacturer_code",
+        "plant_summer_capacity_mw",
+        "water_source",
+    }
+    if fgd.report_date.min() >= pd.Timestamp("2011-01-01T00:00:00"):
+        expected_cols = set(fgd.columns) - fast_run_null_cols
+    else:
+        expected_cols = set(fgd.columns)
+    pudl.validate.no_null_cols(fgd, cols=expected_cols)
+    return AssetCheckResult(passed=True)
+
+
+@asset_check(asset=_core_eia860__fgd_equipment, blocking=True)
+def fgd_cost_discrepancy_check(fgd):
+    """Costs should sum to cost_total.
+
+    To allow for *some* data quality errors we assert that costs ~=
+    total cost at least 99% of the time (with a 1% acceptable
+    discrepancy).
+    """
+
+    def sum_to_target_rate(sum_cols, target, threshold):
```

Comment:
non-blocking: similar to above - "these columns sum to this other column" seems like a nice thing to streamline, eventually.